crates/gitbutler-tauri/src/open.rs

```diff
@@ -0,0 +1,69 @@
+use crate::error::Error;
+use anyhow::anyhow;
+use regex::Regex;
+use std::env;
+use tracing::instrument;
+
+pub fn open_that(path: &str) -> Result<(), Error> {
+    let re = Regex::new(r"^((https://)|(http://)|(mailto:)|(vscode://)|(vscodium://)).+").unwrap();
+    if !re.is_match(path) {
+        return Err(anyhow!("Invalid path format").into());
+    }
+
+    fn clean_env_vars(var_names: &[&str]) -> Vec<(String, String)> {
+        let mut cleaned_vars = Vec::new();
+        for var_name in var_names {
+            if let Ok(var_value) = env::var(var_name) {
```

Comment:
Just as a note: ideally one will always work with `env::var_os()` when paths are involved as UTF-8 can't always represent what's representable on the filesystem. It makes everything harder, too, so I wouldn't bother here also because we know these paths are probably just ASCII :D.
However, when dealing with paths in the belly of the application, one cannot resort to this.